airflow/utils/db.py

```diff
@@ -1557,7 +1557,9 @@ def __setstate__(self, state: Any) -> None:
         self._session = get_current_task_instance_session()
 
     def __bool__(self) -> bool:
-        return check_query_exists(self._select_asc, session=self._session)
+        if check := check_query_exists(self._select_asc, session=self._session) is not None:
```

Comment:
After a second, though, shouldn't we just handle it in `check_query_exists`? That function states it should return `bool`

Reply:
Yep, you're right. I had misremembered and thought `check_query_exists` was a sqlalchemy function, but you're right. The fix should be there.